improve cds.auth (#392)
cap-js/docs: node.js/authentication.md

})
```

> Alternatively, you can also use the sealed instance `cds.User.privileged` directly, i.e., `const user = cds.User.privileged`.

Author: ```suggestion
Alternatively, you can also use the ready-to-use instance `cds.User.privileged` directly, i.e., `const user = cds.User.privileged`.
```

### cds.**User.Anonymous** <i> class </i> { #anonymous-user }

Class `cds.User.Anonymous` allows you to instantiate an anonymous user (`const user = new cds.User.Anonymous`), for example in a [custom authentication](#custom) implementation.

> Alternatively, you can also use the sealed instance `cds.User.anonymous` directly, i.e., `const user = cds.User.anonymous`.

Reviewer: It's the first time we mention a "sealed instance" outside of the changelog. Is that probably a detail we skipped until now by intent?

Author: i don't know about intent, as we don't have any others, afaik. but we can also say something like "ready-to-use instance"

> Alternatively, you can also use the sealed instance `cds.User.anonymous` directly, i.e., `const user = cds.User.anonymous`.

Author: ```suggestion
Alternatively, you can also use the ready-to-use instance `cds.User.anonymous` directly, i.e., `const user = cds.User.anonymous`.
```

### cds.**User.default** { #default-user }

> If a request couldn't be authenticated, for example due to a missing authorization header, the built-in authentication strategies assign the  default user to the request, i.e., `req.user = new cds.User.default`, ensuring that `req.user` is always set.

Reviewer 2: ```suggestion
If a request couldn't be authenticated, for example due to a missing authorization header, the framework will use `cds.User.default` as fallback.
```

}
```

> Essentially, custom authentication middlewares must do two things. First, they _must_ [fulfill the `req.user` contract](#cds-user) by assigning an instance of `cds.User` or a look-alike to the incoming request at `req.user`. Use [cds.User.anonymous](#anonymous-user) as the fallback option. Second, if running in a multitenant environment, `req.tenant` must be set to a string identifying the tenant that is addressed by the incoming request.

Author: ```suggestion
Essentially, custom authentication middlewares must do two things:

First, they _must_ [fulfill the `req.user` contract](#cds-user) by assigning an instance of `cds.User` or a look-alike to the incoming request at `req.user`. Use [new cds.User.default](#default-user) as the fallback option.

Second, if running in a multitenant environment, `req.tenant` must be set to a string identifying the tenant that is addressed by the incoming request.
```

}
```

Essentially, custom authentication middlewares must do two things:

> First, they _must_ [fulfill the `req.user` contract](#cds-user) by assigning an instance of `cds.User` or a look-alike to the incoming request at `req.user`. Use [new cds.User.default](#default-user) as the fallback option.

Reviewer 2: ```suggestion
First, they _must_ [fulfill the `req.user` contract](#cds-user) by assigning an instance of `cds.User` or a look-alike to the incoming request at `req.user`. 
```

`cds.User.default` is always used by default if no `req.user` is set.

[If you want to customize the user ID, please also have a look at this example.](./middlewares#customization-of-req-user){.learn-more}

## Authentication Enforced in Production

> In a productive scenario with an authentication strategy configured, for example the default `xsuaa`, all CAP service endpoints are authenticated by default, regardless of the authorization model. That is, all services without `@restrict` or `@requires` implicitely get `@requires: 'authenticated-user'`. This can be disabled via feature flag `cds.env.requires.auth.restrict_all_services: false`, or by using [mocked authentication](#mocked) explicitly in production.

Author: ```suggestion
In a productive scenario with an authentication strategy configured, for example the default `xsuaa`, all CAP service endpoints are authenticated by default, regardless of the authorization model. That is, all services without `@restrict` or `@requires` implicitely get `@requires: 'authenticated-user'`.

This can be disabled via feature flag `cds.env.requires.auth.restrict_all_services: false`, or by using [mocked authentication](#mocked) explicitly in production.
```

[If you want to customize the user ID, please also have a look at this example.](./middlewares#customization-of-req-user){.learn-more}

## Authentication Enforced in Production

> In a productive scenario with an authentication strategy configured, for example the default `xsuaa`, all CAP service endpoints are authenticated by default, regardless of the authorization model. That is, all services without `@restrict` or `@requires` implicitely get `@requires: 'authenticated-user'`.

Reviewer 3: default is jwt, isn't it?

Author: i had copied from above. fix: https://github.com/cap-js/docs/pull/392/commits/99953e1dedf97f1632e61ca60925b45de9e7a6e1

> In a productive scenario with an authentication strategy configured, for example the default `xsuaa`, all CAP service endpoints are authenticated by default, regardless of the authorization model. That is, all services without `@restrict` or `@requires` implicitely get `@requires: 'authenticated-user'`.

Author: ```suggestion
In a productive scenario with an authentication strategy configured, for example the default `jwt`, all CAP service endpoints are authenticated by default, regardless of the authorization model. That is, all services without `@restrict` or `@requires` implicitely get `@requires: 'authenticated-user'`.
```
